src/Sparrow/Utils/NativeMemory.cs

```diff
@@ -71,7 +73,7 @@ public bool IsThreadAlive()
             public ThreadStats()
             {
                 _threadInstance = Thread.CurrentThread;
-                Id = _threadInstance.ManagedThreadId;
+                Id = Interlocked.Increment(ref _uniqueThreadId);                    
```

Comment:
Need to also have the `ManagedThreadId` here.

Reply:
Rename maybe to `InternalId` and we need to check every place we are using this and probably change bunch of those to `ThreadId`

Reply:
That's too risky move, I think

Reply:
But maybe we should rename it to InternalId, in order for the code to be more correct

Reply:
@maximburyak in https://github.com/ravendb/ravendb/pull/8473/commits/0804bbc5ca48fca2564df7293a2c4dd72370f0ba you introduced `ManagedThreadId` but note that nobody uses it right now. The issue is that `Id` filed, which was the managed thread id, was used in multiple places. I think that all of them should use `ManagedThreadId ` after changes made in this PR.

I'm for renaming `ThreadStats.Id` to `ThreadStats.InternalId`

Reply:
Some correction, maybe not all of them but at least _some of them_ e.g.

- https://github.com/ravendb/ravendb/blob/cd1c8edf84f957153e9442e73248c8e889e33427/src/Raven.Server/Utils/ThreadsUsage.cs#L71

- https://github.com/ravendb/ravendb/blob/cd1c8edf84f957153e9442e73248c8e889e33427/src/Raven.Server/Documents/Handlers/Debugging/MemoryDebugHandler.cs#L287

Reply:
@arekpalinski for debug stuff, we want to use the ManagedThreadId, for any other place, where we want to identify the thread uniquely, we need to use the InternalId

Reply:
Makes sense. So we just need to fix debug stuff usages.

Reply:
done